port/osx/omrosdump.c

```diff
@@ -461,11 +463,17 @@ omrdump_create(struct OMRPortLibrary *portLibrary, char *filename, char *dumpTyp
 		kr = task_set_bootstrap_port(mach_task_self(), special_port);
 		if (KERN_SUCCESS != kr) {
 			mach_error("failed set special port:\n", kr);
-			return kr;
+			goto done;
 		}
 	} /* 0 == child_pid */
 
-	return kr;
+done:
+	/* check the core file has been created in the right directory, or the current if that fails */
+	if (KERN_SUCCESS == kr) {
+		return ((0 == access(filename, F_OK)) || (0 == access(corefile_name, F_OK))) ? 0 : 1;
+	} else {
+		return 1;
+	}
```

Comment:
I think the logic here needs to change. I don't think `filename` is allowed to be `NULL`, so the test on line 416 should change.

If `filename` is empty, the generated name should be written to that buffer.

So this should be something like:
```
	if (KERN_SUCCESS == kr) {
		/* check that the core file has been created */
		if (0 == access(filename, F_OK)) {
			return 0;
		}
		/* check for the core file in the current directory */
		if ((NULL != lastSep) && (0 == access(lastSep + 1, F_OK))) {
			return 0;
		}
	}
	return 1;
```

Reply:
If the caller specified a (possibly empty) file name without a slash, then `lastSep` would be `NULL` in which case `lastSep + 1` would not be a reasonable address to pass to `access()`.